### PR TITLE
Reactive toolbar w.r.t. width

### LINKED ISSUE #445: Toolbar Not Fully Accessible When JupyterCad Editor Width Is Reduced
When the JupyterCad editor's width is reduced beyond a certain point, the toolbar becomes partially inaccessible, making it difficult to use essential buttons. This issue impacts user experience, particularly when working in smaller screen resolutions or when the editor is resized.

### PR DESCRIPTION
**Pull request by arjxn-py**
Should close #445 
Thanks to @brichet

image

### CODE REVIEW

**Comment by arjxn-py** [comment]
Bot please update snapshots.

**Review by martinRenou — APPROVED** [review]
Thanks!

Let's port this to jupytergis too!

**Comment by martinRenou** [comment]
Please update snapshots

**Comment by arjxn-py** [comment]
> Let's port this to jupytergis too!

Sounds Good, happy to do that.

**Comment by martinRenou** [comment]
The visual regression test failure looks legit. It seems our console toggle test fails on:

```typescript
await page.getByRole('button', { name: 'Toggle console' }).click();
```

Probably the selector by role is broken now with the new toolbar?

**Comment by arjxn-py** [comment]
> Probably the selector by role is broken now with the new toolbar?

Helpful hint, I have to look into it

**Comment by arjxn-py** [comment]
It might be because https://github.com/jupytercad/JupyterCAD/blob/main/ui-tests/tests/ui.spec.ts-snapshots/JCAD-Console-linux.png has less width?

And Toggle console button is not accessible without clicking on three dots in that case.

Screenshot 2024-10-09 at 17 43 47

**Comment by arjxn-py** [comment]
Oh, i think test need to be fixed in another PR and has to be in `main` before this one?

File: ui-tests/tests/ui.spec.ts [hunk 1]
```diff
@@ -358,7 +358,11 @@ test.describe('UI Test', () => {
   test.describe('Console activation test', () => {
     test('should open console', async ({ page }) => {
       await page.goto();
-      await page.getByLabel('notebook content').getByText('CAD File').click();
+      await page
+        .getByLabel('notebook content')
+        .getByText('New JCAD File')
```
Comment (martinRenou) [inline ui-tests/tests/ui.spec.ts:363]:
```suggestion
        .getByText('CAD File')
```

**Comment by martinRenou** [comment]
Triggering the CI

**Pull request merged**